### ISSUE TITLE
Relisten is crashing when loading the attributed credits text in the settings view

### ISSUE THREAD

**Issue by farktronix** [RelistenNet/relisten-ios#154]
I think I broke something in the Swift 4.2 conversion. Making a bug so we don't push an update with this crash in place.

```
2018-10-08 09:02:20.155077-0700 Relisten[95287:7611984] <NSATSTypesetter: 0x600002415800>: Exception -[Swift._EmptyArrayStorage _getValue:forType:]: unrecognized selector sent to instance 0x10dd39cb0 raised during typesetting layout manager <ASLayoutManager: 0x600002d36700>
    1 containers, text backing has 77 characters
    Currently holding 77 glyphs.
    Glyph tree contents:  77 characters, 77 glyphs, 1 nodes, 64 node bytes, 512 storage bytes, 576 total bytes, 7.48 bytes per character, 7.48 bytes per glyph
    Layout tree contents:  77 characters, 77 glyphs, 0 laid glyphs, 0 laid line fragments, 1 nodes, 64 node bytes, 0 storage bytes, 64 total bytes, 0.83 bytes per character, 0.83 bytes per glyph, 0.00 laid glyphs per laid line fragment, 0.00 bytes per laid line fragment
, glyph range {0 34}. Ignoring...
2018-10-08 09:02:20.155372-0700 Relisten[95287:7611984] -[Swift._EmptyArrayStorage _getValue:forType:]: unrecognized selector sent to instance 0x10dd39cb0
```

```
(lldb) bt
* thread #27, queue = 'com.apple.root.default-qos', stop reason = breakpoint 1.1
  * frame #0: 0x000000010b689705 libobjc.A.dylib`objc_exception_throw
    frame #1: 0x000000010c10bfa4 CoreFoundation`-[NSObject(NSObject) doesNotRecognizeSelector:] + 132
    frame #2: 0x000000010c0f1fb6 CoreFoundation`___forwarding___ + 1446
    frame #3: 0x000000010c0f3e88 CoreFoundation`__forwarding_prep_0___ + 120
    frame #4: 0x00000001155f673d CoreText`TAttributes::HandleUnderlineAttribute(void const*, TAttributes::AttributeHandlerContext*) + 35
    frame #5: 0x0000000115aaa7d3 UIFoundation`-[NSAttributeDictionary __apply:context:] + 115
    frame #6: 0x00000001155f5d6d CoreText`TAttributes::ParseAttributes(__CFDictionary const*, __CTFont const*, void const*&, TInlineVector<std::__1::function<void (TAttributes&)>, 30ul>&, std::__1::pair<bool, LigatureSetting>*) + 79
    frame #7: 0x00000001155f56af CoreText`TAttributes::ApplyFont(__CFDictionary const*, __CTFont const*, void const**) + 101
    frame #8: 0x00000001155f551d CoreText`TAttributes::TAttributes(__CFDictionary const*) + 133
    frame #9: 0x00000001155d132a CoreText`TRunEncoder::Encode(_CTGlyphStorage*, CFRange, unsigned int, __CFDictionary const*) + 88
    frame #10: 0x000000011559e44f CoreText`CTRunCreateMutableRunsWithStorageAndOptions + 186
    frame #11: 0x0000000115a9c75e UIFoundation`-[NSATSGlyphStorage createCTTypesetter] + 837
    frame #12: 0x0000000115a96806 UIFoundation`-[NSATSTypesetter _ctTypesetter] + 310
    frame #13: 0x0000000115aa148c UIFoundation`-[NSATSLineFragment layoutForStartingGlyphAtIndex:characterIndex:minPosition:maxPosition:lineFragmentRect:] + 82
    frame #14: 0x0000000115a977a3 UIFoundation`-[NSATSTypesetter _layoutLineFragmentStartingWithGlyphAtIndex:characterIndex:atPoint:renderingContext:] + 3336
    frame #15: 0x0000000115a99455 UIFoundation`-[NSATSTypesetter layoutParagraphAtPoint:] + 181
    frame #16: 0x0000000115affaad UIFoundation`-[NSTypesetter _layoutGlyphsInLayoutManager:startingAtGlyphIndex:maxNumberOfLineFragments:maxCharacterIndex:nextGlyphIndex:nextCharacterIndex:] + 3041
    frame #17: 0x0000000115b0157e UIFoundation`-[NSTypesetter layoutCharactersInRange:forLayoutManager:maximumNumberOfLineFragments:] + 242
    frame #18: 0x0000000115a99f74 UIFoundation`-[NSATSTypesetter layoutCharactersInRange:forLayoutManager:maximumNumberOfLineFragments:] + 488
    frame #19: 0x0000000115a801a1 UIFoundation`-[NSLayoutManager(NSPrivate) _fillLayoutHoleForCharacterRange:desiredNumberOfLines:isSoft:] + 1136
    frame #20: 0x0000000115a81d58 UIFoundation`-[NSLayoutManager(NSPrivate) _fillLayoutHoleAtIndex:desiredNumberOfLines:] + 210
    frame #21: 0x0000000115a92a8f UIFoundation`_NSFastFillAllLayoutHolesUpToEndOfContainerForGlyphIndex + 1189
    frame #22: 0x0000000115ac4518 UIFoundation`-[NSLayoutManager ensureLayoutForTextContainer:] + 339
    frame #23: 0x0000000105fa6d33 AsyncDisplayKit`::__34-[ASTextKitTailTruncater truncate]_block_invoke(.block_descriptor=0x000070000e042688, layoutManager=0x0000600002d36700, textStorage=0x00007fc0794899c0, textContainer=0x000060000237c360) at ASTextKitTailTruncater.mm:160
    frame #24: 0x0000000105f90ea9 AsyncDisplayKit`::-[ASTextKitContext performBlockWithLockedTextKitComponents:](self=0x0000600001ff3600, _cmd="performBlockWithLockedTextKitComponents:", block=0x0000000105fa6c30)(NSLayoutManager *, NSTextStorage *, NSTextContainer *)) at ASTextKitContext.mm:79
    frame #25: 0x0000000105fa6bcf AsyncDisplayKit`::-[ASTextKitTailTruncater truncate](self=0x0000600000997440, _cmd="truncate") at ASTextKitTailTruncater.mm:157
    frame #26: 0x0000000105fa0f75 AsyncDisplayKit`::-[ASTextKitRenderer _calculateSize](self=0x000060000261d1e0, _cmd="_calculateSize") at ASTextKitRenderer.mm:145
    frame #27: 0x0000000105fa0226 AsyncDisplayKit`::-[ASTextKitRenderer initWithTextKitAttributes:constrainedSize:](self=0x000060000261d1e0, _cmd="initWithTextKitAttributes:constrainedSize:", attributes=0x000070000e042e10, constrainedSize=(width = 414, height = +Inf)) at ASTextKitRenderer.mm:87
    frame #28: 0x0000000105fd8cd7 AsyncDisplayKit`rendererForAttributes(attributes=ASTextKitAttributes @ 0x000070000e042e10, constrainedSize=(width = 414, height = +Inf)) at ASTextNode.mm:133
    frame #29: 0x0000000105fdaa7f AsyncDisplayKit`::-[ASTextNode _locked_rendererWithBounds:](self=0x00007fc07a92c600, _cmd="_locked_rendererWithBounds:", bounds=(origin = (x = 0, y = 0), size = (width = 414, height = +Inf))) at ASTextNode.mm:347
    frame #30: 0x0000000105fdb4b3 AsyncDisplayKit`::-[ASTextNode calculateSizeThatFits:](self=0x00007fc07a92c600, _cmd="calculateSizeThatFits:", constrainedSize=(width = 414, height = +Inf)) at ASTextNode.mm:393
    frame #31: 0x0000000105e045d6 AsyncDisplayKit`::-[ASDisplayNode calculateLayoutThatFits:](self=0x00007fc07a92c600, _cmd="calculateLayoutThatFits:", constrainedSize=(min = (width = 0, height = 0), max = (width = 414, height = +Inf))) at ASDisplayNode.mm:1231
    frame #32: 0x0000000105e03f04 AsyncDisplayKit`::-[ASDisplayNode calculateLayoutThatFits:restrictedToSize:relativeToParentSize:](self=0x00007fc07a92c600, _cmd="calculateLayoutThatFits:restrictedToSize:relativeToParentSize:", constrainedSize=(min = (width = 0, height = 0), max = (width = 414, height = +Inf)), size=ASLayoutElementSize @ 0x000070000e0437e0, parentSize=(width = 414, height = NaN)) at ASDisplayNode.mm:1182
    frame #33: 0x0000000105dbdd77 AsyncDisplayKit`::-[ASDisplayNode(self=0x00007fc07a92c600, _cmd="layoutThatFits:parentSize:", constrainedSize=(min = (width = 0, height = 0), max = (width = 414, height = +Inf)), parentSize=(width = 414, height = NaN)) layoutThatFits:parentSize:](ASSizeRange, CGSize) at ASDisplayNode+Layout.mm:89
    frame #34: 0x0000000105f481f0 AsyncDisplayKit`crossChildLayout(child=0x00007fc07946f118, style=0x000070000e0445c0, stackMin=0, stackMax=+Inf, crossMin=414, crossMax=414, parentSize=(width = 414, height = NaN)) at ASStackUnpositionedLayout.mm:73
    frame #35: 0x0000000105f47f91 AsyncDisplayKit`::___ZL43layoutItemsAlongUnconstrainedStackDimensionRNSt3__16vectorI21ASStackLayoutSpecItemNS_9allocatorIS1_EEEERK22ASStackLayoutSpecStylebRK11ASSizeRange6CGSizeb_block_invoke(.block_descriptor=0x000070000e044010, i=3) at ASStackUnpositionedLayout.mm:697
    frame #36: 0x0000000105f47c4d AsyncDisplayKit`dispatchApplyIfNeeded(iterationCount=4, forced=NO, work=0x0000000105f47d20) block_pointer) at ASStackUnpositionedLayout.mm:92
    frame #37: 0x0000000105f422d7 AsyncDisplayKit`layoutItemsAlongUnconstrainedStackDimension(items=size=4, style=0x000070000e0445c0, concurrent=NO, sizeRange=0x000070000e044750, parentSize=(width = 414, height = NaN), useOptimizedFlexing=NO) at ASStackUnpositionedLayout.mm:692
    frame #38: 0x0000000105f40ea1 AsyncDisplayKit`ASStackUnpositionedLayout::compute(children=size=4, style=0x000070000e0445c0, sizeRange=0x000070000e044750, concurrent=NO) at ASStackUnpositionedLayout.mm:734
    frame #39: 0x0000000105f2f856 AsyncDisplayKit`::-[ASStackLayoutSpec calculateLayoutThatFits:](self=0x00007fc07b821220, _cmd="calculateLayoutThatFits:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = +Inf))) at ASStackLayoutSpec.mm:155
    frame #40: 0x0000000105e904a4 AsyncDisplayKit`::-[ASLayoutSpec calculateLayoutThatFits:restrictedToSize:relativeToParentSize:](self=0x00007fc07b821220, _cmd="calculateLayoutThatFits:restrictedToSize:relativeToParentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = +Inf)), size=ASLayoutElementSize @ 0x000070000e044920, parentSize=(width = 414, height = NaN)) at ASLayoutSpec.mm:87
    frame #41: 0x0000000105e90237 AsyncDisplayKit`::-[ASLayoutSpec layoutThatFits:parentSize:](self=0x00007fc07b821220, _cmd="layoutThatFits:parentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = +Inf)), parentSize=(width = 414, height = NaN)) at ASLayoutSpec.mm:87
    frame #42: 0x0000000105f481f0 AsyncDisplayKit`crossChildLayout(child=0x0000600003f5b180, style=0x000070000e045330, stackMin=0, stackMax=+Inf, crossMin=414, crossMax=414, parentSize=(width = 414, height = NaN)) at ASStackUnpositionedLayout.mm:73
    frame #43: 0x0000000105f47f91 AsyncDisplayKit`::___ZL43layoutItemsAlongUnconstrainedStackDimensionRNSt3__16vectorI21ASStackLayoutSpecItemNS_9allocatorIS1_EEEERK22ASStackLayoutSpecStylebRK11ASSizeRange6CGSizeb_block_invoke(.block_descriptor=0x000070000e044d80, i=0) at ASStackUnpositionedLayout.mm:697
    frame #44: 0x0000000105f47bd3 AsyncDisplayKit`dispatchApplyIfNeeded(iterationCount=1, forced=NO, work=0x0000000105f47d20) block_pointer) at ASStackUnpositionedLayout.mm:85
    frame #45: 0x0000000105f422d7 AsyncDisplayKit`layoutItemsAlongUnconstrainedStackDimension(items=size=1, style=0x000070000e045330, concurrent=NO, sizeRange=0x000070000e0454c0, parentSize=(width = 414, height = NaN), useOptimizedFlexing=NO) at ASStackUnpositionedLayout.mm:692
    frame #46: 0x0000000105f40ea1 AsyncDisplayKit`ASStackUnpositionedLayout::compute(children=size=1, style=0x000070000e045330, sizeRange=0x000070000e0454c0, concurrent=NO) at ASStackUnpositionedLayout.mm:734
    frame #47: 0x0000000105f2f856 AsyncDisplayKit`::-[ASStackLayoutSpec calculateLayoutThatFits:](self=0x00007fc07bd77780, _cmd="calculateLayoutThatFits:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308))) at ASStackLayoutSpec.mm:155
    frame #48: 0x0000000105e904a4 AsyncDisplayKit`::-[ASLayoutSpec calculateLayoutThatFits:restrictedToSize:relativeToParentSize:](self=0x00007fc07bd77780, _cmd="calculateLayoutThatFits:restrictedToSize:relativeToParentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308)), size=ASLayoutElementSize @ 0x000070000e045690, parentSize=(width = 414, height = 1.7976931348623157E+308)) at ASLayoutSpec.mm:87
    frame #49: 0x0000000105e90237 AsyncDisplayKit`::-[ASLayoutSpec layoutThatFits:parentSize:](self=0x00007fc07bd77780, _cmd="layoutThatFits:parentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308)), parentSize=(width = 414, height = 1.7976931348623157E+308)) at ASLayoutSpec.mm:87
    frame #50: 0x0000000105e4e609 AsyncDisplayKit`::-[ASInsetLayoutSpec calculateLayoutThatFits:restrictedToSize:relativeToParentSize:](self=0x000060000286e220, _cmd="calculateLayoutThatFits:restrictedToSize:relativeToParentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308)), size=ASLayoutElementSize @ 0x000070000e045cc0, parentSize=(width = 414, height = 1.7976931348623157E+308)) at ASInsetLayoutSpec.mm:110
    frame #51: 0x0000000105e90237 AsyncDisplayKit`::-[ASLayoutSpec layoutThatFits:parentSize:](self=0x000060000286e220, _cmd="layoutThatFits:parentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308)), parentSize=(width = 414, height = 1.7976931348623157E+308)) at ASLayoutSpec.mm:87
    frame #52: 0x0000000105e900ac AsyncDisplayKit`::-[ASLayoutSpec layoutThatFits:](self=0x000060000286e220, _cmd="layoutThatFits:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308))) at ASLayoutSpec.mm:87
    frame #53: 0x0000000105e04f58 AsyncDisplayKit`::-[ASDisplayNode calculateLayoutThatFits:](self=0x00007fc07a0f4800, _cmd="calculateLayoutThatFits:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308))) at ASDisplayNode.mm:1283
    frame #54: 0x0000000105e03f04 AsyncDisplayKit`::-[ASDisplayNode calculateLayoutThatFits:restrictedToSize:relativeToParentSize:](self=0x00007fc07a0f4800, _cmd="calculateLayoutThatFits:restrictedToSize:relativeToParentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308)), size=ASLayoutElementSize @ 0x000070000e0464f0, parentSize=(width = 414, height = 1.7976931348623157E+308)) at ASDisplayNode.mm:1182
    frame #55: 0x0000000105dbdd77 AsyncDisplayKit`::-[ASDisplayNode(self=0x00007fc07a0f4800, _cmd="layoutThatFits:parentSize:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308)), parentSize=(width = 414, height = 1.7976931348623157E+308)) layoutThatFits:parentSize:](ASSizeRange, CGSize) at ASDisplayNode+Layout.mm:89
    frame #56: 0x0000000105dbd0cc AsyncDisplayKit`::-[ASDisplayNode(self=0x00007fc07a0f4800, _cmd="layoutThatFits:", constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308))) layoutThatFits:](ASSizeRange) at ASDisplayNode+Layout.mm:65
    frame #57: 0x0000000105d9ed07 AsyncDisplayKit`::-[ASDataController _layoutNode:withConstrainedSize:](self=0x0000600002018dc0, _cmd="_layoutNode:withConstrainedSize:", node=0x00007fc07a0f4800, constrainedSize=(min = (width = 414, height = 0), max = (width = 414, height = 1.7976931348623157E+308))) at ASDataController.mm:202
    frame #58: 0x0000000105d9e708 AsyncDisplayKit`::__58-[ASDataController _allocateNodesFromElements:completion:]_block_invoke(.block_descriptor=0x0000600000819000, i=0) at ASDataController.mm:185
    frame #59: 0x0000000105db5351 AsyncDisplayKit`__ASDispatchApply_block_invoke(.block_descriptor=0x0000600000818dc0) at ASDispatch.m:32
    frame #60: 0x000000010e6855d1 libdispatch.dylib`_dispatch_call_block_and_release + 12
    frame #61: 0x000000010e68663e libdispatch.dylib`_dispatch_client_callout + 8
    frame #62: 0x000000010e68923b libdispatch.dylib`_dispatch_queue_override_invoke + 1439
    frame #63: 0x000000010e697028 libdispatch.dylib`_dispatch_root_queue_drain + 351
    frame #64: 0x000000010e6979cd libdispatch.dylib`_dispatch_worker_thread2 + 130
    frame #65: 0x000000010ea756ee libsystem_pthread.dylib`_pthread_wqthread + 619
    frame #66: 0x000000010ea75415 libsystem_pthread.dylib`start_wqthread + 13
```

**Comment by farktronix** [comment]
https://stackoverflow.com/questions/52408760/swift-4-2-swift-emptyarraystorage-getvaluefortype-unrecognized-selector

**Comment by farktronix** [comment]
Fixed with d6bc651
